src/librustc/middle/dead.rs

```diff
@@ -172,6 +172,7 @@ impl<'a, 'tcx> MarkSymbolVisitor<'a, 'tcx> {
                     | hir::ItemKind::Ty(..)
                     | hir::ItemKind::Static(..)
                     | hir::ItemKind::Existential(..)
+                    | hir::ItemKind::Impl(..)
                     | hir::ItemKind::Const(..) => {
                         intravisit::walk_item(self, &item);
                     }
```

Comment:
Remove `_ => ()` and replace with "exhaustive" match to avoid future problems?

Reply:
I'm not actually sure why this is not
```rust
_ => {
    intravisit::walk_item(self, &item);
}
```
If that doesn't break anything, then `walk_item` can be moved out of the `match`.